File: Assignment 1/first.py
from OpenGL.GL import *
import glfw


glfw.init()

window = glfw.create_window(1280, 1000, "Hello World", None, None)

glfw.make_context_current(window)

while not glfw.window_should_close(window):
    glfw.poll_events()
    glClear(GL_COLOR_BUFFER_BIT)

    glColor3f(0.0, 50.0, 100.0)

    glBegin(GL_POLYGON)

    glVertex2f(0, 0.5)
    glVertex2f(0.5, -0.25)
    glVertex2f(-0.5, -0.25)

    glEnd()
    glFlush()

    glfw.swap_buffers(window)

glfw.terminate()

File: Assignment 1/first.cpp
#include <GLFW/glfw3.h>

int main(void)
{
    GLFWwindow *window;

    /* Initialize the library */
    if (!glfwInit())
        return -1;

    /* Create a windowed mode window and its OpenGL context */
    window = glfwCreateWindow(1280, 1000, "Hello World", NULL, NULL);
    if (!window)
    {
        glfwTerminate();
        return -1;
    }

    /* Make the window's context current */
    glfwMakeContextCurrent(window);

    /* Loop until the user closes the window */
    while (!glfwWindowShouldClose(window))
    {
        /* Poll for and process events */
        glfwPollEvents();

        /* Render here */
        glClear(GL_COLOR_BUFFER_BIT);

        glColor3f(0.0, 50.0, 100.0);

        glBegin(GL_TRIANGLES);

        glVertex2f(0, 0.5);
        glVertex2f(0.5, -0.25);
        glVertex2f(-0.5, -0.25);

        glEnd();
        glFlush();

        /* Swap front and back buffers */
        glfwSwapBuffers(window);
    }

    glfwTerminate();
    return 0;
}
